kubectl-cloudflow/verify/blueprint.go

```diff
@@ -0,0 +1,1007 @@
+	var streamletRefWithClassNameUpdated StreamletRef
+	var streamletRefWithMetadataUpdated StreamletRef
+	for _, streamlRef := range b.streamlets {
+		if streamlRef.name == streamletRef {
+			if className != nil {
+				streamletRefWithClassNameUpdated = StreamletRef{
+					name:      streamlRef.name,
+					className: streamlRef.className,
+					verified:  streamlRef.verified,
+					metadata: streamlRef.metadata,
+					imageId: streamlRef.imageId,
+				}
+			} else {
+				streamletRefWithClassNameUpdated = streamlRef
+			}
+
+			if metadata != nil {
+				streamletRefWithMetadataUpdated = StreamletRef{
+					name:      streamletRefWithClassNameUpdated.name,
+					className: streamletRefWithClassNameUpdated.className,
+					verified:  streamletRefWithClassNameUpdated.verified,
+					metadata: metadata,
+					imageId: streamletRefWithClassNameUpdated.imageId,
+				}
+			} else {
+				streamletRefWithMetadataUpdated = streamletRefWithClassNameUpdated
+			}
+
+			ret := Blueprint(b)
+
+			streamletsToAppend := []StreamletRef{}
+			for _, streamlet := range b.streamlets {
+			 if streamlet.name != streamlRef.name {
+			 	streamletsToAppend = append(streamletsToAppend, streamlet)
+			 }
+			}
+			ret.streamlets = streamletsToAppend
+			ret.streamlets = append(ret.streamlets, streamletRefWithMetadataUpdated)
+			return ret.verify()
+		}
+	}
+
+	if className != nil {
+		return b.use(StreamletRef{
+			name: streamletRef,
+			className: *className,
+			metadata: metadata,
+		})
+	} else{
+		return b
+	}
+}
+
+func getDurationFromConfig(config string) (duration time.Duration, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			var ok bool
+			err, ok = r.(error)
+			if !ok {
+				err = fmt.Errorf("Parsing duration error: %v", r)
+			}
+		}
+	}()
+	return configuration.ParseString(fmt.Sprintf("value=%s", config)).GetTimeDuration("value", time.Nanosecond), nil
+}
+
+func getMemorySizeFromConfig(config string) (size *big.Int, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			var ok bool
+			err, ok = r.(error)
+			if !ok {
+				err = fmt.Errorf("Parsing memory size error: %v", r)
+			}
+		}
+	}()
+	return configuration.ParseString(fmt.Sprintf("value=%s", config)).GetByteSize("value"), nil
+}
+
+func (b Blueprint) verifyConfigParameters(streamletDescriptors []StreamletDescriptor) []BlueprintProblem {
+	var invalidConfigParametersKeyProblems, invalidDefaultValueOrPatternProblems, duplicateConfigParametersKeysFound []BlueprintProblem
+	var keys []string
+	for _, desc := range streamletDescriptors {
+		for _, configParam := range desc.ConfigParameters {
+			keys = append(keys, configParam.Key)
+
+			if !CheckFullPatternMatch(configParam.Key, ConfigParameterKeyPattern) {
+				invalidConfigParametersKeyProblems = append(invalidConfigParametersKeyProblems, InvalidConfigParameterKeyName{className: desc.ClassName, keyName: configParam.Key})
+			}
+
+			switch configParam.Type {
```

Comment:
No `default` case ?

Reply:
I committed because the default action is do nothing at the Scala side. But I will added it for completeness.